x/checkers/keeper/msg_server_create_game.go

```diff
@@ -2,16 +2,46 @@ package keeper
 
 import (
 	"context"
+	"strconv"
 
 	sdk "github.com/cosmos/cosmos-sdk/types"
+	rules "github.com/xavierlepretre/checkers/x/checkers/rules"
 	"github.com/xavierlepretre/checkers/x/checkers/types"
 )
 
 func (k msgServer) CreateGame(goCtx context.Context, msg *types.MsgCreateGame) (*types.MsgCreateGameResponse, error) {
 	ctx := sdk.UnwrapSDKContext(goCtx)
 
-	// TODO: Handling the message
-	_ = ctx
+	nextGame, found := k.Keeper.GetNextGame(ctx)
+	if !found {
+		panic("NextGame not found")
+	}
+	newIndex := strconv.FormatUint(nextGame.IdValue, 10)
+	creator, err := sdk.AccAddressFromBech32(msg.Creator)
+	if err != nil {
+		return nil, types.ErrInvalidCreator
+	}
+	red, err := sdk.AccAddressFromBech32(msg.Red)
+	if err != nil {
+		return nil, types.ErrInvalidRed
+	}
+	black, err := sdk.AccAddressFromBech32(msg.Black)
+	if err != nil {
+		return nil, types.ErrInvalidBlack
```

Comment:
@marbar3778 Is there a way to forward the `err` I got as part of the `ErrInvalidBlack`? That would greatly improve the information flow.